src/mapnik_vector_tile.cpp

```diff
@@ -58,10 +59,122 @@
 #include "vector_tile_backend_pbf.hpp"
 #include <mapnik/datasource_cache.hpp>
 #include <mapnik/hit_test_filter.hpp>
+
+
 #include <google/protobuf/io/coded_stream.h>
+#include "vector_tile.pb.h"
 
 namespace detail {
 
+inline void add_tile(std::string & buffer, vector_tile::Tile const& tile)
+{
+    std::string new_message;
+    if (!tile.SerializeToString(&new_message))
+    {
+        /* LCOV_EXCL_START */
+        throw std::runtime_error("could not serialize new data for vt");
+        /* LCOV_EXCL_END */
+    }
+    if (!new_message.empty())
+    {
+        buffer.append(new_message.data(),new_message.size());
+    }
+}
+
+inline vector_tile::Tile get_tile(std::string const& buffer)
+{
+    vector_tile::Tile tile;
+    if (!tile.ParseFromArray(buffer.data(), buffer.size()))
+    {
+        throw std::runtime_error("could not parse as proto from buf");
```

Comment:
No test coverage here, if no possible way to test please wrap with LCOV as shown above.